json-endpoints/class.wpcom-json-api-get-comments-tree-v1-1-endpoint.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+new WPCOM_JSON_API_Get_Comments_Tree_v1_1_Endpoint ( array(
+	'description' => 'Get a comments tree for site.',
+	'min_version' => '1.1',
+	'max_version' => '1.1',
+	'group'       => 'comments-tree',
+	'stat'        => 'comments-tree:1',
+
+	'method'      => 'GET',
+	'path'        =>  '/sites/%s/comments-tree',
+	'path_labels' => array(
+		'$site'   => '(int|string) Site ID or domain',
+	),
+	'query_parameters' => array(
+		'status' => '(string) Filter returned comments based on this value (allowed values: all, approved, pending, trash, spam).'
```

Comment:
I know this has already been discussed, but I'd like to raise this objection again. 🙂 

We use `pending` here as this is what we do on the front end, but [comment list uses `unapproved`](https://github.com/Automattic/jetpack/blob/master/json-endpoints/class.wpcom-json-api-list-comments-endpoint.php#L122).

Wouldn't it make sense to update the [status mapping](https://github.com/Automattic/jetpack/blob/master/json-endpoints/class.wpcom-json-api-get-comments-tree-endpoint.php#L131) to:

```php
if ( 'pending' === $status || 'unapproved' === status ) {
  return '0';
}
```

in order to allow for both `pending` and `unapproved`?

Reply:
where is this problematic? I don't have any personal objection, but as long as it's well documented is it causing any issue? seems like the terminology is closer to the UI

Reply:
It's closer to _our_ UI, but fails in not being consistent with a well established endpoint.

If I've always used `unapproved` when fetching a comment list, I'm inclined to keep using `unapproved` in any comment context, including comment tree, without the need of checking the documentation because I wouldn't expect any differences.

Reply:
I actually like this idea, it's weirded me out for a bit too. I'll do it as separate PRs to this though, as there will be no harm in adding it to the existing endpoint. 

Dotcom: D7602-code
JP: TBD

Reply:
at this point would you prefer just standardizing on `unapproved` and removing `pending`?

Reply:
If I had to choose one or the other, I'd prefer swapping in `unapproved` instead of keeping just `pending` – but would _really_ prefer having both. What is your thinking against it?

Reply:
against keeping both? consistency and complexity

for keeping both? easier right now, `pending` more closely matches how we talk about that status

Reply:
I vote for replacing it with `unapproved` that's already used in comments list. Consistency trumps everything. :)

Reply:
@vindl We've added support fo `unapproved` in 683f150 (but kept `pending`).